### PR TITLE
[621] Improve Software Management Architecture doc

### PR DESCRIPTION
**Pull request by rina23q**
## Proposed changes

Filled contents in our Software Management Architecture doc.

Question: How much contents should we want? We also have specification doc. I wrote bare minimum.

## Types of changes

What types of changes does your code introduce to `thin-edge.io`?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [x] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: docs/src/architecture/software-management.md [hunk 1]
```diff
@@ -5,45 +5,31 @@ Software Management operates end to end from a cloud down to the OS of your devi
 
 ## Software management components
 
-Software Management uses following components to perform software operations:
+Software Management uses following the 3 components to perform software operations:
```
Comment (makr11st) [inline docs/src/architecture/software-management.md:8]:
```suggestion
Software Management uses the following 3 components to perform software operations:
```

File: docs/src/architecture/software-management.md [hunk 2]
```diff
@@ -5,45 +5,31 @@ Software Management operates end to end from a cloud down to the OS of your devi
...
-The installation will add `systemd` service `tedge-agent.service` and new user specific to the agent (`tedge-agent`).
-As some of the operations may require `root` permissions or `sudo` access it is advised that the tedge-agent user is added to the `sudo` group which will allow it to execute elevated commands.
-
-You can do with following command:
-
-```shell
-sudo usermod -aG sudo tedge-agent
-```
+The **Tedge Agent** addresses cloud-agnostic software management operations along with the **Cloud Mapper**,
+e.g. listing current installed software list, software update, software removal.
```
Comment (makr11st) [inline docs/src/architecture/software-management.md:21]: Shouldn't the Cloudmapper be removed from this sentence?
```suggestion
The **Tedge Agent** addresses cloud-agnostic software management operations e.g. listing current installed software list, software update, software removal.
```

File: docs/src/architecture/software-management.md [hunk 3]
```diff
@@ -5,45 +5,31 @@ Software Management operates end to end from a cloud down to the OS of your devi
...
+Also, the Tedge Agent calls the **SM Plugin(s)** to execute an action defined by a received operation.
 
-To start the agent you can do:
+The key point is that the Tedge Agent is always generic in cloud platforms and software types.
```
Comment (makr11st) [inline docs/src/architecture/software-management.md:24]: How about this?
```suggestion
The key point is that the **Tedge Agent** is always generic in cloud platforms and software types and **Tedge Mapper** handles cloud specific actions.
```

File: docs/src/architecture/software-management.md [hunk 4]
```diff
@@ -5,45 +5,31 @@ Software Management operates end to end from a cloud down to the OS of your devi
...
-As some of the operations may require `root` permissions or `sudo` access it is advised that the tedge-agent user is added to the `sudo` group which will allow it to execute elevated commands.
-
-You can do with following command:
-
-```shell
-sudo usermod -aG sudo tedge-agent
-```
+The **Tedge Agent** addresses cloud-agnostic software management operations along with the **Cloud Mapper**,
+e.g. listing current installed software list, software update, software removal.
+Also, the Tedge Agent calls the **SM Plugin(s)** to execute an action defined by a received operation.
```
Comment (makr11st) [inline docs/src/architecture/software-management.md:22]:
```suggestion
Also, the Tedge Agent calls an **SM Plugin(s)** to execute an action defined by a received operation.
```

**Review by makr11st — COMMENTED** [review]
I suppose it is right amount of knowledge in here as we already have other tutorials and docs regarding the topic.
One thing I would like to see here is probably some diagram with interactions between components (some people take better visualisations), maybe we don't have to create new ones but at least give links to particular places in design/spec or other documents?

File: docs/src/architecture/software-management.md (hunk 3 above)
Comment (rina23q) [reply docs/src/architecture/software-management.md:23]: A bit correction:
```suggestion
The key points are that the **Tedge Agent** is always generic in cloud platforms and software types, and **Cloud Mapper** handles cloud-specific actions.
```

File: docs/src/architecture/README.md [hunk 5]
```diff
@@ -24,5 +24,5 @@ This is an optional feature, and the components are free to also use cloud speci
 3. The mapper processes are responsible for translating the canonical data format into cloud specific messages and vice versa.
 
 
-![Overview](./thin-edge-overview.png)
+![Overview](./images/thin-edge-overview.png)
```
Comment (rina23q) [inline docs/src/architecture/README.md:27]: @makr11st  Like this? Then we don't need to add a new image.

```suggestion
You can find the diagrams which explain how those 3 components interact with each other from [Software Management Speficication](https://thin-edge.github.io/thin-edge.io-specs/software-management/sm-agent.html).
```

Comment (makr11st) [reply docs/src/architecture/README.md:27]: Yes, but I would leave this diagram as it is (general overview) and add the other link in software-management.md

**Review by makr11st — APPROVED** [review]

File: docs/src/architecture/README.md (hunk 5 above)
Comment (rina23q) [reply docs/src/architecture/README.md:27]: Oh no! Here's not what I meant!!! :D:D:D

File: docs/src/architecture/software-management.md [hunk 6]
```diff
@@ -5,45 +5,30 @@ Software Management operates end to end from a cloud down to the OS of your devi
 
 ## Software management components
 
-Software Management uses following components to perform software operations:
+Software Management uses the following 3 components to perform software operations:
+**Cloud Mapper**, **Agent**, and **Software Management Plugin**.
+
+![Software management components](./images/software-management.png)
```
Comment (rina23q) [inline docs/src/architecture/software-management.md:11]:
```suggestion
You can find the diagrams which explain how those 3 components interact with each other from [Software Management Agent Specification](https://thin-edge.github.io/thin-edge.io-specs/software-management/sm-agent.html).
```

**Pull request merged**